### ISSUE TITLE
keras==3.7.0 - Attention(... return_attention_scores=True) does not work

### ISSUE THREAD

**Issue by Furkan-rgb** [keras-team/keras#20680]
With keras==3.7.0, the following snippet:

```python
import keras
from keras import layers

attention = layers.Attention()

x = keras.Input(shape=(3, 5))
y = keras.Input(shape=(3, 5))

out_attn, out_attn_scores = attention(x, y, return_attention_scores=True)
```
Throws:
```python
Traceback (most recent call last):
  File "attention_repro.py", line 10, in <module>
    out_attn, out_attn_scores = attention(x, y, return_attention_scores=True)
  File "venv\lib\site-packages\keras\src\backend\common\keras_tensor.py", line 121, in __iter__
    raise NotImplementedError(
NotImplementedError: Iterating over a symbolic KerasTensor is not supported.
```
Because return_attention_scores seems to have no effect, and the second output is not returned. The attention.call() method seems right, it's just that the __call__ doesn't seem to construct the output tuple correctly.

**Comment by Furkan-rgb** [comment]
It is the exact same issue as #19303

**Comment by Furkan-rgb** [comment]
Made a PR with fix: #20684

**Comment by Furkan-rgb** [comment]
Fixed #20684
